tests/test_modeling_tf_common.py

```diff
@@ -380,65 +366,191 @@ def test_pt_tf_model_equivalence(self):
+            if names == "past_key_values":
+                return
+
+            if type(tfo) == tuple:
+                self.assertEqual(type(pto), tuple)
+                self.assertEqual(len(tfo), len(pto))
+                if type(names) in [tuple, list]:
+                    for to, po, name in zip(tfo, pto, names):
+                        check_outputs(to, po, model_class, names=name)
+                elif type(names) == str:
+                    for idx, (to, po) in enumerate(zip(tfo, pto)):
+                        check_outputs(to, po, model_class, names=f"{names}_{idx}")
+            elif isinstance(tfo, tf.Tensor):
+                self.assertTrue(isinstance(pto, torch.Tensor))
+
+                tfo = tfo.numpy()
+                pto = pto.numpy()
+
+                tf_nans = np.copy(np.isnan(tfo))
+                pt_nans = np.copy(np.isnan(pto))
+
+                pto[tf_nans] = 0
+                tfo[tf_nans] = 0
+                pto[pt_nans] = 0
+                tfo[pt_nans] = 0
+
+                max_diff = np.amax(np.abs(tfo - pto))
+                self.assertLessEqual(max_diff, 1e-5)
 
-            tf_hidden_states = tfo[0].numpy()
-            pt_hidden_states = pto[0].numpy()
+        for model_class in self.all_model_classes:
 
-            tf_nans = np.copy(np.isnan(tf_hidden_states))
-            pt_nans = np.copy(np.isnan(pt_hidden_states))
+            config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
 
-            pt_hidden_states[tf_nans] = 0
-            tf_hidden_states[tf_nans] = 0
-            pt_hidden_states[pt_nans] = 0
-            tf_hidden_states[pt_nans] = 0
+            # Output all for aggressive testing
+            config.output_hidden_states = True
+            # Pure convolutional models have no attention
+            # TODO: use a better and general criteria
+            if "TFConvNext" not in model_class.__name__:
+                config.output_attentions = True
+
+            # TODO: remove this block once the large negative value for attention masks is fixed.
+            for k in ["attention_mask", "encoder_attention_mask", "decoder_attention_mask"]:
+                if k in inputs_dict:
+                    attention_mask = inputs_dict[k]
+                    # (make sure no all 0s attention masks - to avoid failure at this moment)
+                    attention_mask = tf.ones_like(attention_mask, dtype=tf.int32)
+                    # (make the first sequence with all 0s attention mask -> to demonstrate the issue)
+                    # (this will fail for `TFWav2Vec2Model`)
+                    # attention_mask = tf.concat(
+                    #     [
+                    #         tf.zeros_like(attention_mask[:1], dtype=tf.int32),
+                    #         tf.cast(attention_mask[1:], dtype=tf.int32)
+                    #     ],
+                    #     axis=0
+                    # )
```

Comment:
To clean up?

Reply:
Will remove it later - this part will be activated and used for a follow PR to ensure everything works well.

(dealing with the `1e-4`, `1e-9`, `1e-30` values for attn mask. Currently, different values across frameworks will introduce larger errors, when a sequence has all 0s attention mask ).